### ISSUE TITLE
AnimatedIcons should have Cupertino-specific versions

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#17854]
The Cupertino lib should include AnimatedIcons similar to the ones provided for Material that animated from the hamburger menu to the Cupertino style back button and vice versa. 

https://docs.flutter.io/flutter/material/AnimatedIcons-class.html, specifically talking about `menu_arrow` and `arrow_menu`, which should have equivalents in Cupertino.

/cc @xster @amirh

**Comment by goderbauer** [comment]
Once we have that, we should use it in the Search Demo in the gallery.

**Comment by xster** [comment]
Do you have a link to where it does it in the iOS spec? It might be a material feature rather (toggling between a hamburger and an iOS back button)

**Comment by goderbauer** [comment]
Yeah, it's part of how Material should feel on iOS. Did I tag it incorrectly?

**Comment by goderbauer** [comment]
Hehe. OK. Something should provide it. 😝
